src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -288,22 +311,36 @@ async def call_api_partial(page):
                     original_request=request,
                     form_data=form_data,
                     filename=file.file_name,
-                    page=page,
+                    pdf_chunk=pdf_chunk,
                     limiter=limiter,
                 )
 
+                # Immediately delete request to save memory
+                del response._request  # pylint: disable=protected-access
+                response._request = None  # pylint: disable=protected-access
+
+                # If we get 200, dump the contents to a file and return the path
+                temp_dir = self.tempdirs[operation_id]
+                if response.status_code == 200:
+                    temp_file_name = f"{temp_dir.name}/{uuid.uuid4()}.json"
```

Comment:
Is there a risk that the customer using the client doesn't have permissions to write files in the current process they're running?

Reply:
Good point. But at the same time `unstructured` OS uses temp files heavily. Maybe making the temp dir parametrized via env could resolve this?

Reply:
yeah would guess for our own environments it's safe but a customer could hit an error. maybe we could fall back to using memory if we don't have write access (not sure on implementation there), or we could expose a path in the client that the caller sets up explicitly for temporary storage maybe?

Reply:
I've added a feature flag in partition parameters (feat disabled by default)